clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/CloudMetricStatistics.groovy

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import groovy.transform.Immutable
+
+interface CloudMetricStatistics {
+  /**
+   * Unit of measurement for all datapoints; should be the plural form of the unit if applicable, e.g. "bytes"
+   */
+  String unit
+
+  /**
+   * List of statistical datapoints; at least one statistic (average, sum, sampleCount, minimum, maximum) should be
+   * populated, as well as the timestamp
+   */
+  List<CloudMetricDatapoint> datapoints
+
+  @Immutable
+  @JsonInclude(JsonInclude.Include.NON_EMPTY)
+  static class CloudMetricDatapoint {
+    Date timestamp
+    Double average
+    Double sum
+    Double sampleCount
```

Comment:
More FYI than anything else.
I'm not too familiar with Google Cloud Monitoring yet.

Browsing the API, it looks like the model is a bit different.
Points in a time series (which looks like this corresponds to) have a start/end time rather than instantaneous timestamp. Values are not necessarily Double -- they could be any primitive type (string, bool, int64, double) as well as distributions. For purposes of spinnakers, needs double might be sufficient (certainly ok as a starting point). However this interface is restrictive to which metrics you can tap into.

Distributions are bucketed value ranges containing bounds and a count (bounds are Double). So looks like they can comply to this interface. Except there is an overflow and underflow bucket, so I guess those values would get thrown away. Not sure how that plays in practice. We'd have to compute the average (which is reasonable) and leave other statistical possibilities on the table for now.